src/Microsoft.OpenApi/Validations/ValidationRule.cs

```diff
@@ -29,7 +29,7 @@ public abstract class ValidationRule
     /// Class containing validation rule logic for <see cref="IOpenApiElement"/>.
     /// </summary>
     /// <typeparam name="T"></typeparam>
-    public class ValidationRule<T> : ValidationRule where T : IOpenApiElement
+    public class ValidationRule<T> : ValidationRule
```

Comment:
Ugh. This is because JsonSchema doesn't implement this interface.  Hmm, that seems like it is going to cause us problems.